src/components/Shared/SafeNumericInput/SafeNumericInput.tsx

```diff
@@ -1,14 +1,61 @@
 import * as React from "react";
-import {NumericInput} from "@blueprintjs/core";
+import {NumericInput, INumericInputProps} from "@blueprintjs/core";
 
-export class SafeNumericInput extends React.PureComponent<any> {
-    safeHandleValueChanged = (valueAsNumber: number, valueAsString: string) => {
-        if (this.props.onValueChange && isFinite(valueAsNumber) && (!this.props.min || this.props.min <= valueAsNumber) && (!this.props.max || this.props.max >= valueAsNumber)) {
-            this.props.onValueChange(valueAsNumber, valueAsString);
+export interface SafeNumericInputProps extends INumericInputProps {
+    integerOnly?: boolean;
+    handleBlur? (ev: React.FocusEvent<HTMLInputElement>) : void;
+    handleKeyDown? (ev: React.KeyboardEvent<HTMLInputElement>) : void;
+}
+
+export class SafeNumericInput extends React.Component<SafeNumericInputProps> {
+    private static minorStepSize = 0.001;
+
+    safeHandleValueChanged = (valueAsNumber: number, valueAsString: string, inputElement: HTMLInputElement) => {
+        if (this.props.integerOnly) {
+            const roundValue = Math.ceil(valueAsNumber);
+            if (isFinite(roundValue)) {
+                valueAsNumber = roundValue;
+                valueAsString = roundValue.toString();
+                inputElement.value = roundValue.toString();
+            }
+        }
+
+        if (this.props.onValueChange && isFinite(valueAsNumber) && (!this.props.min || this.props.min <= valueAsNumber) && (!this.props.max || this.props.max >= valueAsNumber)) {    
+            this.props.onValueChange(valueAsNumber, valueAsString, inputElement);
         }
     };
 
     render() {
-        return <NumericInput {...this.props} onValueChange={this.safeHandleValueChanged}/>;
+        return (
+            <NumericInput
+                allowNumericCharactersOnly={this.props.allowNumericCharactersOnly}
```

Comment:
@panchyo0 can we just use `{...props}` for most of these?

Reply:
{...this.props} will result a warning message in console `React does not recognize the prop on a DOM element`

Reply:
just rename `handleKeyDown` and `handleBlur` to `onKeyDown` and `onBlur` respectively, and that will get rid of the warnings

Reply:
thanks